.github/workflows/ci.yml

```diff
@@ -31,99 +34,68 @@ jobs:
         with:
           submodules: true
 
-      - name: Setup Go
-        uses: actions/setup-go@cdcb36043654635271a94b9a6d1392de5bb323a7 # v5
+      - name: Setup Earthly
+        uses: earthly/actions-setup@v1
         with:
-          go-version: ${{ env.GO_VERSION }}
-
-      - name: Find the Go Build Cache
-        id: go
-        run: echo "cache=$(make go.cachedir)" >> $GITHUB_OUTPUT
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          version: ${{ env.EARTHLY_VERSION }}
 
-      - name: Cache the Go Build Cache
-        uses: actions/cache@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4
+      - name: Login to DockerHub
+        uses: docker/login-action@e92390c5fb421da1463c202d546fed0ec5c39f20 # v3
+        if: env.DOCKER_USR != ''
         with:
-          path: ${{ steps.go.outputs.cache }}
-          key: ${{ runner.os }}-build-check-diff-${{ hashFiles('**/go.sum') }}
-          restore-keys: ${{ runner.os }}-build-check-diff-
+          username: ${{ secrets.DOCKER_USR }}
+          password: ${{ secrets.DOCKER_PSW }}
 
-      - name: Cache Go Dependencies
-        uses: actions/cache@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@e92390c5fb421da1463c202d546fed0ec5c39f20 # v3
         with:
-          path: .work/pkg
-          key: ${{ runner.os }}-pkg-${{ hashFiles('**/go.sum') }}
-          restore-keys: ${{ runner.os }}-pkg-
-
-      - name: Download Go Modules
-        run: make modules.download modules.check
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Check Diff
-        run: make check-diff
+      - name: Generate Files
+        run: earthly --ci +generate
 
-  detect-noop:
```

Comment:
is this intentionally removing the no-op detection or is that also to be reimplemented?

Reply:
I did remove it intentionally. My thinking was that when Earthly has a hot cache it'll handle this automatically. It only rebuilds if the code changes, so changing a Markdown file or similar will be a no-op at the Earthly level.

TBD as to whether that will work though. We'd need to setup [remote caching](https://docs.earthly.dev/docs/caching/caching-via-registry#explicit-cache-advanced) for Earthly to be able to handle it automatically.

There's also a handful of relatively long running CI jobs that don't run in Earthly (yet?) - e.g. fuzz tests and CodeQL. We'd need to either get those running in Earthly, or reintroduce detect-noop. I'd personally prefer to get everything running in Earthly if only to keep things simple and consistent.